rig/machine_control/machine_controller.py

```diff
@@ -1657,51 +1661,153 @@ def get_p2p_routing_table(self, x, y):
         return table
 
     @ContextMixin.use_contextual_arguments()
+    def get_chip_info(self, x, y):
+        """Get general information about the resources available on a chip.
+
+        Returns
+        -------
+        :py:class:`.ChipInfo`
+            A named tuple indicating the number of working cores, the states of
+            all working cores, the set of working links and the size of the
+            largest free block in SDRAM and SRAM.
+        """
+        info = self._send_scp(x, y, 0, SCPCommands.info, expected_args=3)
+        num_cores = info.arg1 & 0x1F
+        core_states = [consts.AppState(c)
+                       for c in struct.unpack("<18B", info.data)]
+        working_links = set(link for link in Links
+                            if (info.arg1 >> (8 + link)) & 1)
```

Comment:
This set-comprehension? I'm not sure but the new `get_system_info` runs faster than the old `get_machine`. I'd be very disappointed if anything but the network round-trips were dominating runtime at the moment. It would be worth investigating appropriate ways to make all those SCP commands get sent in parallel in the future...